infrastructure/oss/vulnerability_count.go

```diff
@@ -75,27 +85,71 @@ func (v *vulnerabilityCountImpl) ProcessVulnerabilityCount(
 			highestIssueSeverity = issue.Severity
 			vc.mostSevereVulnerabilityId = issue.ID
 		}
-	}
 
-	for _, vulnerabilityCount := range vulnerabilityCounts {
-		for _, vci := range vulnerabilityCount {
-			v.c.Logger().Debug().
-				Str("method", method).
-				Str("filePath", vci.FilePath).
-				Str("range", vci.Range.String()).
-				Int("total", vci.total).
-				Str("mostSevereVulnerabilityId", vci.mostSevereVulnerabilityId).
-				Msg("processing vulnerability count")
-			err := processor(*vci)
-			if err != nil {
-				v.c.Logger().Err(err).Str("method", method).Msg("error processing vulnerability count")
-				return err
-			}
-		}
+		logger.Debug().Any("vulnerability count", vc).Msgf("updated")
 	}
-	return nil
+
+	logger.Debug().Int("count", len(vulnerabilityCounts)).
+		Msgf("created vulnerability counts")
+	return vulnerabilityCounts, nil
 }
 
 func NewVulnerabilityCountService(c *config.Config) VulnerabilityCountService {
 	return &vulnerabilityCountImpl{c: c}
 }
+
+func (oss *Scanner) addVulnerabilityCountsAsInlineValuesToCache(issues []snyk.Issue) {
+	c := config.CurrentConfig()
+	logger := c.Logger().With().Str("method", "addVulnerabilityCountsAsInlineValuesToCache").Logger()
+	vc := NewVulnerabilityCountService(c)
+	counts, err := vc.DetermineVulnerabilityCounts(issues)
+	if err != nil {
+		logger.Err(err).Msg("couldn't get vulnerability counts")
+		oss.errorReporter.CaptureError(err)
+	}
+	for _, r := range counts {
+		for _, vci := range r {
+			value := oss.toInlineValueAndAddToCache(vci)
+			logger.Debug().
+				Any("vci", value).
+				Msg("added to cache")
+		}
+	}
+}
+
+func (oss *Scanner) getDisplayTextForVulnerabilityCount(vci *VulnerabilityCountInformation) string {
+	text := fmt.Sprintf(
+		"Vulnerabilities: %d | Critical: %d, High: %d, Medium: %d, Low: %d | Most Severe: %s",
+		vci.total,
+		vci.severityCounts[snyk.Critical],
+		vci.severityCounts[snyk.High],
+		vci.severityCounts[snyk.Medium],
+		vci.severityCounts[snyk.Low],
+		vci.mostSevereVulnerabilityId,
+	)
+	return text
+}
+
+func (oss *Scanner) toInlineValueAndAddToCache(vci *VulnerabilityCountInformation) snyk.InlineValue {
+	values := oss.inlineValues[vci.FilePath]
```

Comment:
suggestion: it is a bit tricky to read this code as it uses values and value. maybe rename values to cachedValues, to better differentiate.